com.unity.perception/Runtime/GroundTruth/Labelers/KeyPointLabeler.cs

```diff
@@ -309,7 +309,7 @@ void ProcessEntity(Labeling labeledEntity)
                     keyPoints[idx].index = idx;
                     keyPoints[idx].x = loc.x;
                     keyPoints[idx].y = loc.y;
-                    keyPoints[idx].state = 1;
+                    keyPoints[idx].state = 2;
```

Comment:
Why would this be 2 instead of 1 or 0? Based on the comments in the Keypoint class definition, this state value can only be either 1 or 0 based on whether the Keypoint is present or not. Actually, I'm not sure why this isn't a boolean. There's another line right above this line that also sets state to 2, and I'm not sure why that it.

Reply:
So according to Steve, 1 means the joint is present but not visible and 2 means present and visible. These are based on COCO. 0 means not present. 

Steve said 1 is not actually implemented yet (future work). When you add a joint manually, it should always be 2 since it exists.